integration/inspektor-gadget/run_insecure_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+)
+
+func TestRunInsecure(t *testing.T) {
+	ns := GenerateTestNamespaceName("test-run-insecure")
+
+	t.Parallel()
+
+	commandsPreTest := []*Command{
+		CreateTestNamespaceCommand(ns),
+		PodCommand("registry", "docker.io/library/registry:2", ns, "", ""),
+		WaitUntilPodReadyCommand(ns, "registry"),
+	}
+
+	RunTestSteps(commandsPreTest, t)
+
+	t.Cleanup(func() {
+		commands := []*Command{
+			DeleteTestNamespaceCommand(ns),
+		}
+		RunTestSteps(commands, t, WithCbBeforeCleanup(PrintLogsFn(ns)))
+	})
+
+	registryIP := GetTestPodIP(t, ns, "registry")
+
+	// copy gadget image to insecure registry
+	orasCpCmds := []*Command{
+		{
+			Cmd: fmt.Sprintf("kubectl run -n %s copier --image ghcr.io/oras-project/oras:v1.1.0 --restart=Never -- copy --to-plain-http %s/trace_open:%s %s:5000/trace_open:%s",
+				ns, *gadgetRepository, *gadgetTag, registryIP, *gadgetTag),
+		},
+		WaitUntilPodReadyCommand(ns, "copier"),
+		{
+			// Wait for the image to be available in the registry
+			// TODO: Ideally it shouldn't be a delay, but there is no an easy way to
+			// wait for a completed command using kubectl run
+			Cmd: "sleep 2",
```

Comment:
Couldn't we use something like `kubectl wait --for=Succeeded` ? (Ref for `Succeeded` https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase )

Or we add `--rm` to the `kubectl run` command and wait for the pod deletion`

Reply:
> Couldn't we use something like kubectl wait --for=Succeeded ?

Unfortunately it doesn't work. We can only wait on any of these conditions -> https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

I decided to change it and use a job, that's a better fit for this use case. Can you check it again please?

Reply:
See https://github.com/inspektor-gadget/inspektor-gadget/pull/2261